core/test/nuxeo-document.test.js

```diff
@@ -129,4 +130,72 @@ suite('nuxeo-document', () => {
       expect(document.path).to.be.eq('/path/something');
     });
   });
+
+  suite('when setting document view or download action', () => {
+    test('should set view and download data', async () => {
+      const document = await fixture(
+        html`
+          <nuxeo-document doc-id="something"></nuxeo-document>
+        `,
+      );
+      let data = {
+        properties: {
+          'file:content': {
+            appLinks: [],
+            data: 'file1.jpeg?changeToken=13-0',
+            digest: '2e7d1a1ba7018c048bebdf1d07481ee3',
+            digestAlgorithm: 'MD5',
+            encoding: null,
+            length: '5763',
+            'mime-type': 'image/jpeg',
+            name: 'kitten1 (4).jpeg',
+          },
+        },
+        schemas: [
+          {
+            name: 'video',
+          },
+        ],
+      };
+      document.documentData = data;
+      document.setDocumentViewDownloadProp();
+      expect(document.documentData.properties['file:content'].viewData).to.eq(
+        'file1.jpeg?changeToken=13-0&clientReason=view',
+      );
+      expect(document.documentData.properties['file:content'].downloadData).to.eq(
+        'file1.jpeg?changeToken=13-0&clientReason=download',
+      );
+      data = {
+        properties: {
+          'vid:transcodedVideos': [
+            {
+              content: {
+                data: 'vid1.mp4?changeToken=9-0',
+                'mime-type': 'video/mp4',
+              },
+              info: {
+                width: 66,
+                height: 66,
+                format: 'jpeg',
+              },
+              name: 'vid1.mp4',
+            },
+          ],
+        },
+        schemas: [
+          {
+            name: 'video',
+          },
+        ],
+      };
+      document.documentData = data;
+      document.setDocumentViewDownloadProp();
```

Comment:
no need to have separate initialize of data and calling of setDocumentViewDownloadProp. Kindly merge with above data